arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorizedArrowReader.java

```diff
@@ -240,6 +247,12 @@ private void allocateFieldVector(boolean dictionaryEncodedVector) {
             this.readType = ReadType.LONG;
             this.typeWidth = (int) BigIntVector.TYPE_WIDTH;
             break;
+          case TIME_MICROS:
```

Comment:
as discussed offline do we want to add support for MILLIs?

Reply:
It looks like Parquet's `TimeWriter` only writes micros in https://github.com/nastra/iceberg/blob/50f4ecca7711e69f63589fea828d26230fac8d59/parquet/src/main/java/org/apache/iceberg/data/parquet/BaseParquetWriter.java#L256, so I think the answer would be that we don't need to handle `TIME_MILLIS`

Reply:
I will look into supporting `TIME_MILLIS` as a follow-up tomorrow

Reply:
after some investigation, supporting `TIME_MILLIS` might be a bit more involved. I opened https://github.com/apache/iceberg/issues/2755